src/pubsub/index.js

```diff
@@ -21,6 +21,18 @@ const {
   verifySignature
 } = require('./message/sign')
 
+/**
+ * @typedef {any} Libp2p
+ * @typedef {import('peer-id')} PeerId
+ * @typedef {import('bl')} BufferList
+ * @typedef {import('../stream-muxer/types').MuxedStream} MuxedStream
+ * @typedef {import('../connection/connection')} Connection
+ * @typedef {import('./message').RPC} RPC
+ * @typedef {import('./message').SubOpts} RPCSubOpts
+ * @typedef {import('./message').Message} RPCMessage
+ * @typedef {import('./signature-policy').SignaturePolicyType} SignaturePolicyType
+ */
+
 /**
  * @typedef {Object} InMessage
  * @property {string} [from]
```

Comment:
It is really dependent on the signature policy. Have a look into the `validate` function. The same for other places like the `verifySignature` function, which `validate` only calls if it has a signature. I will add an error there to avoid potential future usages of the function though